File: A1023.java
//long long 存不下 20 位正整数，用 java 大数

import java.math.BigInteger;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner cin = new Scanner(System.in);
		int [] cnt1 = new int[10];
		int [] cnt2 = new int[10];
		BigInteger a = cin.nextBigInteger();
		BigInteger tmp = a;
		BigInteger m = BigInteger.valueOf(10);
		BigInteger l = BigInteger.valueOf(0);
		while(tmp.compareTo(l) != 0) {
			BigInteger t = tmp.mod(m);
			int dex1 = t.intValue();
			cnt1[dex1]++;
			tmp = tmp.divide(m);
		}
		BigInteger k = BigInteger.valueOf(2);
		a = a.multiply(k);
		tmp = a;
		while(tmp.compareTo(l) != 0) {
			BigInteger t = tmp.mod(m);
			int dex2 = t.intValue();
			cnt2[dex2]++;
			tmp = tmp.divide(m);
		}
		int flag = 0;
		for(int i = 0; i < 10; i++) {
			if(cnt1[i] != cnt2[i]) {
				flag = 1;
				break;
			}
		}
		if(flag == 1) System.out.println("No");
		else System.out.println("Yes");
		System.out.println(a);
	}
}

File: A1023.cpp
//大数相乘

#include <iostream>
using namespace std;

int x[10], y[10];

string mul(string a){
    int lena = a.size();
    int cnt = 0;
    for(int i = lena - 1; i >= 0; i--) {
        int res = (a[i] - '0') * 2 + cnt;
        if(res >= 10) {
            a[i] = res % 10 + '0';
            cnt = res / 10;
        }
        else {
            a[i] = res + '0';
            cnt = 0;
        }
        if(i == 0 && cnt) {
            a = char(cnt + '0') + a;
        }
    }
    return a;
}

int main()
{
    string s;
    cin >> s;
    string ans = mul(s);
    int len = ans.size();
    for(int i = 0; i < len; i++) {
        x[int(ans[i] - '0')]++;
    }
    int lens = s.size();
    for(int i = 0; i < lens; i++) {
        y[int(s[i] - '0')]++;
    }
    int flag = 1;
    for(int i = 0; i < 10; i++) {
        if(x[i] != y[i]) {
            flag = 0;
            break;
        }
    }
    if(flag) cout << "Yes" << endl;
    else cout << "No" << endl;
    cout << ans << endl;
    return 0;
}